store/copr/region_cache.go

```diff
@@ -75,59 +75,59 @@ func (l *LocationKeyRanges) splitKeyRangesByBuckets() []*LocationKeyRanges {
 
 	ranges := l.Ranges
 	loc := l.Location
+
+	// Sometimes, there may be some delay between region information and buckets information within the region.
+	// It may cause some gaps between the region startKey and first buckets key and region endKey and the last buckets key.
+	// To easy the implementation of this function, we add region startKey if it is less than the first bucket key and add region endKey if
+	// it is larger than the last bucket key. Without the addition, loc.LocateBucket may return nil which can cause troubles.
+	if len(loc.Buckets.Keys) == 0 || bytes.Compare(loc.StartKey, loc.Buckets.Keys[0]) < 0 {
+		loc.Buckets.Keys = append([][]byte{loc.StartKey}, loc.Buckets.Keys...)
```

Comment:
The content of the location should not be changed here, it's read-only in the original implementation.  A possible option is to check the `end_key` if `LocateBucket ` returns nil or changing the logic of `LocateBucket `.
There's no mutable constraint in golang, be very careful if you decide to change the content of some input references.